daprdocs/content/en/operations/components/pluggable-components/pluggable-components-discovery.md

```diff
@@ -0,0 +1,212 @@
+
+Follow the steps provided in the [Deploy Dapr on a Kubernetes cluster]({{< ref kubernetes-deploy.md >}}) docs and make sure you have a kubernetes cluster configured with Dapr version 1.9+.
+
+## Step 3: Adding Pluggable Component Container in your Deployments
+
+When running on kubernetes mode, gRPC-based Pluggable Components are side-car containers.
+
+As gRPC-based Pluggable Components are backed by [Unix Domain Sockets][uds], the first step is configuring the deployment spec to ensure that the socket created by your Pluggable Component are accessible by Dapr runtime, to do that, you have to:
+
+1. mount volumes,
+2. hint Dapr where the mounted unix socket volume is and also,
+3. attach such volume to your Pluggable Component container.
+
+Below you can see an example of a Deployment that configures a Pluggable Component:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+  labels:
+    app: app
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: app
+  template:
+    metadata:
+      labels:
+        app: app
+      annotations:
+        dapr.io/unix-domain-socket-path: "/tmp/dapr-components-sockets" ## required, the default path where Dapr will discovery components.
+        dapr.io/app-id: "my-app"
+        dapr.io/enabled: "true"
+        dapr.io/sidecar-listen-addresses: "0.0.0.0"
+    spec:
+      volumes: ## required, the sockets volume
+        - name: dapr-unix-domain-socket
+          emptyDir: {}
+      containers:
+        ### --------------------- YOUR APPLICATION CONTAINER GOES HERE -----------
+        ##
+        ### --------------------- YOUR APPLICATION CONTAINER GOES HERE -----------
+        ### This is the pluggable component container.
+        - name: component
+          volumeMounts: # required, the sockets volume mount
+            - name: dapr-unix-domain-socket
+              mountPath: /dapr-unix-domain-sockets
+          image: YOUR_IMAGE_GOES_HERE:YOUR_IMAGE_VERSION
+          env:
+            - name: DAPR_COMPONENTS_SOCKETS_FOLDER # Tells the component where the sockets should be created.
+              value: /dapr-unix-domain-sockets
+```
+
+Great, do not apply the Deployment yet, let's add one more configuration: The Component Spec.
+
+## Step 4: Declaring a gRPC-based Pluggable Component
+
+gRPC-based Pluggable Componets are defined using the [Component Spec]({{< ref component-schema.md >}}) and its `type` is derived from the socket name (without the file extension).
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: prod-mystore
+spec:
+  type: your_component_type.your_socket_goes_here
+  version: v1
+  metadata:
+scopes:
+  - backend
+```
+
+You should also [scope]({{< ref component-scopes >}}) your component to make sure that only the target configured application will try to connect with the pluggable component since it will only be running in its Deployment, otherwise the runtime will fail when initializing the component.
```

Comment:
```suggestion
[Scope]({{< ref component-scopes >}}) your component to make sure that only the target configured application will try to connect with the pluggable component, since it will only be running in its deployment. Otherwise the runtime will fail when initializing the component.
```